kernel/src/main/scala/me/rexim/morganey/church/ChurchPairConverter.scala

```diff
@@ -16,11 +16,23 @@ object ChurchPairConverter {
       case _ => None
     }
 
-  def encodePair(p: (LambdaTerm, LambdaTerm), cons: String = "x"): LambdaTerm = {
-    val (car, cdr) = p
-    LambdaFunc(LambdaVar(cons),
-      LambdaApp(LambdaApp(LambdaVar(cons), car), cdr)
-    )
+  def encodePair(pair: (LambdaTerm, LambdaTerm)): LambdaTerm = {
+    val (first, second) = pair
+    val x = LambdaVar("x")
+
+    val pairFunc = {
+      val y = LambdaVar("y")
+      val z = LambdaVar("z")
+
+      LambdaFunc(y,
+        LambdaFunc(z,
+          LambdaApp(LambdaApp(z, x), y)))
```

Comment:
Sorry, not worth the effort for me at the moment. It will require to change all of the UTs back because of the lack of #16. I don't mind to discuss different naming approaches for encoded pairs in a separate issue, though. Feel free to create it.